Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -178,9 +202,28 @@ class UIRedactBuilder {
     }
     
     private func shouldIgnore(view: UIView) -> Bool {
-        return SentryRedactViewHelper.shouldUnmask(view) || containsIgnoreClass(type(of: view))
+        return  SentryRedactViewHelper.shouldUnmask(view) || containsIgnoreClass(type(of: view)) || shouldIgnoreParentContainer(view)
     }
-    
+
+    private func shouldIgnoreParentContainer(_ view: UIView) -> Bool {
+        if isRedactContainerClass(type(of: view)) {
+            return false
+        }
+
+        guard let parent = view.superview  else { return false }
+        return isIgnoreContainerClass(type(of: parent))
```

Comment:
m: Try to use `guard` as much as possible for early return.

```suggestion
        guard !isRedactContainerClass(type(of: view)),  let parent = view.superview  else { return false }
        return isIgnoreContainerClass(type(of: parent))
```